persistent-test/src/PersistentTest.hs

```diff
@@ -224,6 +228,25 @@ specs = describe "persistent" $ do
       ps <- selectList [FilterAnd []] [Desc PersonAge]
       assertNotEmpty ps
 
+#ifndef WITH_MONGODB
+  it "LIKE" $ db $ do
+      let p = Person "Shakespeare" 52 Nothing
+      _ <- insert p
+
+      let maltext = "re%"
+      ps <- selectList [likeWithEscape '@' PersonName ("%espea" <> toEscapedLikeText maltext)] []
+      assertEmpty ps
+
+  it "NOT LIKE" $ db $ do
+      let p = Person "Shakespeare" 52 Nothing
+      _ <- insert p
+
+      let maltext = "re%"
+      ps <- selectList [notLikeWithEscape '@' PersonName ("%espea" <> toEscapedLikeText maltext)] []
+      assertNotEmpty ps
+#endif
```

Comment:
The test suite has been revamped to not use CPP to split out backends. I'd prefer to see a new module `LikeTest` that can be used to demonstrate support for `LIKE` on many backends, including MongoDB when that is done.